query-service-impl/src/main/java/org/hypertrace/core/query/service/pinot/QueryRequestToPinotSQLConverter.java

```diff
@@ -282,6 +285,8 @@ private String convertExpression2String(Expression expression, Params.Builder pa
         // COUNT(*) for ORDER BY
         if (functionName.equalsIgnoreCase("COUNT")) {
           return functionName + "(*)";
+        } else if (functionName.startsWith(PERCENTILE_PREFIX) && !PERCENTILE_PREFIX.equals(functionName)) {
```

Comment:
I think abstracting the percentile at this level is correct. This allows us to potentially change storage without changing the QueryService API.
Should we allow both Percentile and ApproxPercentile though as a function? 
We can add it later as long as we can configure function -> implementation

Reply:
The value of config lets you add either Percentile or Approx. That's the config indeed :)